File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    vector<int> vec;
    Solution()
    {
        vec = vector<int>(46,-1);
        vec[0] = vec[1] = 1;
    }
    int climbStairs(int n) 
    {
        if(vec[n] == -1)
            vec[n] = climbStairs(n-1) + climbStairs(n-2);
        return vec[n];
    }
};

File: 70-climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n<=1){
            return 1;
        }
        int dp[] = new int[46];
        dp[0]=1;
        dp[1]=1;
        for(int i=2;i<46;i++){
            dp[i]=dp[i-1]+dp[i-2];
        }
        return dp[n];
    }
}